File: XamarinProject/BindingBindingsDemoApp/MainActivity.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Widget;

namespace BindingBindingsDemoApp
{
    [Activity(Label = "BindingBindingsDemoApp", MainLauncher = true, Icon = "@mipmap/icon")]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            // Get our button from the layout resource,
            // and attach an event to it
            Button button = FindViewById<Button>(Resource.Id.myButton);

            button.Click += delegate {
                var intent = new Intent(this, typeof(Databinding.Android.Librarywithbinding.BindingSampleActivity));
                StartActivity(intent);
            };
        }
    }
}

File: AndroidProject/app/src/main/java/databinding/android/databindingexample/MainActivity.java
package databinding.android.databindingexample;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;

import databinding.android.librarywithbinding.BindingSampleActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button button = this.findViewById(R.id.button);
        button.setOnClickListener(view -> {
            Intent intent = new Intent(getBaseContext(), BindingSampleActivity.class);
            startActivity(intent);
        });
    }
}
